### PR TITLE
chore/enterpriseportal: add test for iam_model

### PR DESCRIPTION
**Pull request by bobheadxi**
Uses the guidance in https://openfga.dev/docs/modeling/testing to craft some rudimentary IAM model tests for Enterprise Portal IAM.

Not automated for now - the model tests must be run manually:

```
go run github.com/openfga/cli/cmd/fga@latest model test --tests='cmd/enterprise-portal/service/iam_model.fga.yml'
```

If we end up changing the model more I'll ask around in dev-infra to see how we should automate this.

## Test plan

CI and:

```
go run github.com/openfga/cli/cmd/fga@latest model test --tests='cmd/enterprise-portal/service/iam_model.fga.yml'
```

### CODE REVIEW

File: cmd/enterprise-portal/service/iam_model.fga.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+tuples:
+- user: user:018d21f2-04a6-7aaf-9f6f-6fc58c4187b9
+  relation: member
+  object: customer_admin:80ca12e2-54b4-448c-a61a-390b1a9c1224
+
+- user: user:617E04BC-905D-4543-94E7-55DCF7F5E92F
+  relation: member
+  object: customer_admin:4CDA7E9F-69D8-4594-9637-F964693ADFE5
+
```
Comment (jac) [inline cmd/enterprise-portal/service/iam_model.fga.yml:17]: don't we need to make it so members of a specific customer admin have access to specific subscriptions?
```suggestion
- user: customer_admin:80ca12e2-54b4-448c-a61a-390b1a9c1224#member
  relation: view
  object: subscription_cody_analytics:80ca12e2-54b4-448c-a61a-390b1a9c1224

- user: customer_admin:4CDA7E9F-69D8-4594-9637-F964693ADFE5#member
  relation: view
  object: subscription_cody_analytics:4CDA7E9F-69D8-4594-9637-F964693ADFE5

```

(is it expected that customer admin objects and subscription_cody_analytics objects have the same IDs?)

File: cmd/enterprise-portal/service/iam_model.fga.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  check:
+    - user: user:018d21f2-04a6-7aaf-9f6f-6fc58c4187b9
+      object: subscription_cody_analytics:80ca12e2-54b4-448c-a61a-390b1a9c1224
+      assertions:
+        view: false
+
+    - user: user:617E04BC-905D-4543-94E7-55DCF7F5E92F
+      object: subscription_cody_analytics:CDA7E9F-69D8-4594-9637-F964693ADFE5
+      assertions:
+        view: false
```
Comment (jac) [inline cmd/enterprise-portal/service/iam_model.fga.yml:52]: the assertions should be they can view, right? (+ a typo in the second subscription id)
```suggestion
        view: true

    - user: user:617E04BC-905D-4543-94E7-55DCF7F5E92F
      object: subscription_cody_analytics:4CDA7E9F-69D8-4594-9637-F964693ADFE5
      assertions:
        view: true
```

File: cmd/enterprise-portal/service/iam_model.fga.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+- name: non-customer_admin members cannot access Cody Analytics
+  check:
+    - user: user:617E04BC-905D-4543-94E7-55DCF7F5E92F
+      object: subscription_cody_analytics:80ca12e2-54b4-448c-a61a-390b1a9c1224
+      assertions:
+        view: false
+
+    - user: user:018d21f2-04a6-7aaf-9f6f-6fc58c4187b9
+      object: subscription_cody_analytics:CDA7E9F-69D8-4594-9637-F964693ADFE5
```
Comment (jac) [inline cmd/enterprise-portal/service/iam_model.fga.yml:62]: same typo
```suggestion
      object: subscription_cody_analytics:4CDA7E9F-69D8-4594-9637-F964693ADFE5
```

File: cmd/enterprise-portal/service/iam_model.fga.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      assertions:
+        view: false
+
+    - user: user:EA71BAB8-09CA-4014-A62F-119B291C62B9 # unknown user
+      object: subscription_cody_analytics:80ca12e2-54b4-448c-a61a-390b1a9c1224
+      assertions:
+        view: false
+
+    - user: user:EA71BAB8-09CA-4014-A62F-119B291C62B9 # unknown user
+      object: subscription_cody_analytics:CDA7E9F-69D8-4594-9637-F964693ADFE5
```
Comment (jac) [inline cmd/enterprise-portal/service/iam_model.fga.yml:72]: same typo
```suggestion
      object: subscription_cody_analytics:4CDA7E9F-69D8-4594-9637-F964693ADFE5
```

**Review by jac — CHANGES_REQUESTED** [review]
this doesn't look correct to me. it was passing but I don't think the assertions were actually correct

_the uuids are painful to review lol_ 😭

**Review by jac — APPROVED** [review]

**Review by unknwon — COMMENTED** [review]
TIL!

**Pull request merged**